src/unix/apple/cpu.rs

```diff
@@ -371,9 +382,33 @@ pub(crate) fn get_vendor_id_and_brand() -> (String, String) {
     (vendor, get_sysctl_str(b"machdep.cpu.brand_string\0"))
 }
 
+pub(crate) unsafe fn get_cpu_arch() -> CpuArch {
+    use std::ffi::CStr;
+    let mut mib: [c_int; 2] = [libc::CTL_HW, libc::HW_MACHINE_ARCH];
+    let mut arch_str: [u8; 32] = [b'\0'; 32];
+
+    if get_sys_value(
+        libc::CTL_HW as _,
+        libc::HW_MACHINE as _,
+        mem::size_of::<[char; 32]>(),
+        &mut arch_str as *mut [u8; 32] as *mut c_void,
```

Comment:
I'm pretty sure you can do:

```suggestion
        arch_str.as_mut_ptr() as *mut _,
```